### PR TITLE
Let render monitor process delete temp file in test

### PR DESCRIPTION
**Pull request by miklosduma**
One of the tests uses a subprocess that creates a temporary file. This fix stops the subprocess in a way that it leaves time to delete the temp file.

### CODE REVIEW

File: sparkl_cli/test/test_put_source_rm.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def test_render_local_monitor(self):
         sleep(2)
 
         # Stop monitoring process
-        monitor_process.terminate()
+        monitor_process.join(timeout=1)
```
Comment (lucafavatellawork) [inline sparkl_cli/test/test_put_source_rm.py]: From https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.terminate
> Terminate the process. On Unix this is done using the SIGTERM signal ...

From https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.join
> If the optional argument timeout is None (the default), the method blocks until the process whose join() method is called terminates. If timeout is a positive number, it blocks at most timeout seconds. Note that the method returns None if its process terminates or if the method times out. Check the process’s exitcode to determine if it terminated.

From PR description:
> One of the tests uses a subprocess that creates a temporary file. This fix stops the subprocess in a way that it leaves time to delete the temp file.

So this (test) code changes semantics. Before, a SIGTERM was sent to the process. Now, it simply waits until the process spontaneous terminates or 1 second elapses - whatever comes earlier. Did you intend to rather to rather do (untested):
```suggestion
        monitor_process.join(timeout=1)
        if monitor_process.exitcode is not None:
            monitor_process.terminate()
```
? (Just checking.)

Comment (miklosduma) [reply sparkl_cli/test/test_put_source_rm.py]: Wait, the exitcode will be None if the process has not terminated yet. So we want to terminate it if it has terminated?

Another option might be to set the process to daemon, so it won't be left orphaned

Comment (lucafavatellawork) [reply sparkl_cli/test/test_put_source_rm.py]: > Wait, the exitcode will be None if the process has not terminated yet. So we want to terminate it if it has terminated?

Correct, my code is wrong.

Comment (miklosduma) [reply sparkl_cli/test/test_put_source_rm.py]: If I go with your suggestion the temp file stays. Cleanest option that both deletes the temp file and leaves no orphans is setting the process to a daemon. Will update the code. Thanks for the heads up! ^ @lucafavatellawork

File: sparkl_cli/test/test_put_source_rm.py [hunk 2]
```diff
@@ -153,21 +154,25 @@ def test_render_local_monitor(self):
...
-            assert "TestCall" in data.read()
+            # Check if the output file was updated
+            with open("tmp.html", 'r') as data:
+                assert "TestCall" in data.read()
 
-        # Get rid of the temporary file
-        os.remove(temp_path)
+        finally:
+            # Interrupt monitor process to stop generator.
+            os.kill(monitor_process.pid, signal.SIGINT)
```
Comment (lucafavatellawork) [inline sparkl_cli/test/test_put_source_rm.py]: From https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions
> The finally clause runs whether or not the try statement produces an exception.

Ok.

**Pull request merged**